### PR TITLE
Fix issue with unmasked logs

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-44767)

## Description
Fixing an issue where a specific pattern was not masked in the logs.
Fixed an issue where an error returned from Google  was fully displayed to the user as-is, exposing the entire message in the War Room and the logs.
Requesting a force-merge due to 2 test playbook that with expired credentials:
GSuiteAdmin-Test
 playbook-GSuiteAdmin-MobileDevices-Test
 A ticket was opened to the lab.
 
## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@JasBeilin Doc review completed.

File: Packs/ApiModules/Scripts/GSuiteApiModule/GSuiteApiModule.py [hunk 1]
```diff
@@ -294,3 +295,32 @@ def strip_dict(args: Dict[str, str]) -> Dict[str, str]:
         :return: Dictionary with whitespaces and empty entries removed.
         """
         return {key: value.strip() for (key, value) in args.items() if value and value.strip()}
+
+
+''' HELPER FUNCTIONS '''
+
+
+def replace_token_text(text: str) -> str:
```
Comment (JasBeilin) [inline Packs/ApiModules/Scripts/GSuiteApiModule/GSuiteApiModule.py:303]: replace with regex, get the prefix as input and move to CSP.

**Review by JasBeilin — CHANGES_REQUESTED** [review]

File: Packs/ApiModules/Scripts/GSuiteApiModule/GSuiteApiModule.py (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/ApiModules/Scripts/GSuiteApiModule/GSuiteApiModule.py]: Done

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -12500,6 +12514,29 @@ def function_runner(func, profiler, signal_event,
...
+
+    :return: None
+    """
+    sensitive_pattern = re.compile(pattern)
+    match = sensitive_pattern.search(text)
+
+    if not match:
+        return
+
+    sensitive_text = match.group(2)
```
Comment (JasBeilin) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:12534]: Lets verify.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py [hunk 3]
```diff
@@ -9994,3 +9994,40 @@ def test_get_engine_base_url(mocker):
     mocker.patch.object(demisto, 'internalHttpRequest', return_value=mock_response)
     res = get_engine_base_url('1111')
     assert res == '11.111.111.33:443'
+
+
+def test_find_and_remove_sensitive_text__found(mocker):
```
Comment (JasBeilin) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython_test.py:9999]: Lets add some cases, this feels lacking. Lets try with a different regex, when the whole message is sensitive, when regex shows more than once, etc.

**Review by JasBeilin — CHANGES_REQUESTED** [review]
waiting for @tkatzir  comment as well.

File: Packs/Base/ReleaseNotes/1_38_4.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+#### Scripts
+
+##### CommonServerPython
+
+Fixed a masking issue in the logs.
```
Comment (JasBeilin) [inline Packs/Base/ReleaseNotes/1_38_4.md:6]:
```suggestion
log improvement.
```

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -12538,6 +12552,32 @@ def function_runner(func, profiler, signal_event,
...
+
+    :return: None
+    """
+    sensitive_pattern = re.compile(pattern)
+    matches = sensitive_pattern.findall(text)
+    if not matches:
+        return
+
+    for match in matches:
+        if isinstance(match, tuple):
```
Comment (JasBeilin) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:12573]: Why using the cases? add a comment explaining whats happening please.

File: Packs/GSuiteAdmin/ReleaseNotes/1_2_5.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Google Workspace Admin
+
+Fixed an issue where an error message from the vendor, was not properly masked in the logs.
```
Comment (JasBeilin) [inline Packs/GSuiteAdmin/ReleaseNotes/1_2_5.md:6]:
```suggestion
Improved error message from the vendor.
```

File: Packs/GSuiteSecurityAlertCenter/ReleaseNotes/1_1_48.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### G Suite Security Alert Center
+
+Fixed an issue where an error message from the vendor, was not properly masked in the logs.
```
Comment (JasBeilin) [inline Packs/GSuiteSecurityAlertCenter/ReleaseNotes/1_1_48.md:6]:
```suggestion
Improved error message from the vendor.```

File: Packs/GoogleCalendar/ReleaseNotes/1_1_50.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Google Calendar
+
+Fixed an issue where an error message from the vendor, was not properly masked in the logs.
```
Comment (JasBeilin) [inline Packs/GoogleCalendar/ReleaseNotes/1_1_50.md:6]:
```suggestion
Improved error message from the vendor.```

File: Packs/GoogleCloudLogging/ReleaseNotes/1_0_18.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Google Cloud Logging
+
+Fixed an issue where an error message from the vendor, was not properly masked in the logs.
```
Comment (JasBeilin) [inline Packs/GoogleCloudLogging/ReleaseNotes/1_0_18.md:6]:
```suggestion
Improved error message from the vendor.```

File: Packs/GoogleDrive/ReleaseNotes/1_3_12.md [hunk 10]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Google Drive
+
+Fixed an issue where an error message from the vendor, was not properly masked in the logs.
```
Comment (JasBeilin) [inline Packs/GoogleDrive/ReleaseNotes/1_3_12.md:6]:
```suggestion
Improved error message from the vendor.```

**Review by JasBeilin — APPROVED** [review]
NIce. I would clarify the comment in the group match to include example for the one word case.
Other than that this looks good.

**Pull request merged**